tutorials/tour/_posts/2017-02-13-mixin-class-composition.md

```diff
@@ -9,28 +9,41 @@ categories: tour
 num: 6
 next-page: anonymous-function-syntax
 previous-page: traits
+prerequisite-knowledge: inheritance, traits, abstract-classes, unified-types
 ---
+Mixin class composition means that a class can reuse the members of one or more mixins (i.e. traits and types).
+
+```tut
+abstract class A {
+  def message: String
+}
+class B extends A {
+  def message = "I'm an instance of class B"
+}
+trait C extends A {
+  def makeMessageLoud = message.toUpperCase()
+}
+class D extends B with C
+
+val d = new D
+d.message  // I'm an instance of class B
+d.makeMessageLoud  // I'M AN INSTANCE OF CLASS B
+```
+Class `D` has a superclass `B` and a mixin `C`. Classes can only have one superclass but many mixins (using the keywords `extends` and `with` respectively). The mixins and the superclass may have the same supertype.
+
+Now let's look at a more interesting example starting with an abstract class:
 
-As opposed to languages that only support _single inheritance_, Scala has a more general notion of class reuse. Scala makes it possible to reuse the _new member definitions of a class_ (i.e. the delta in relationship to the superclass) in the definition of a new class. This is expressed as a _mixin-class composition_. Consider the following abstraction for iterators.
- 
 ```tut
 abstract class AbsIterator {
   type T
   def hasNext: Boolean
   def next: T
 }
 ```
- 
-Next, consider a mixin class which extends `AbsIterator` with a method `foreach` which applies a given function to every element returned by the iterator. To define a class that can be used as a mixin we use the keyword `trait`.
- 
-```tut
-trait RichIterator extends AbsIterator {
-  def foreach(f: T => Unit) { while (hasNext) f(next) }
-}
-```
- 
-Here is a concrete iterator class, which returns successive characters of a given string:
- 
+This abstract iterator class could be used as a parameter type where an iterator is needed.
+
+Next, we'll implement a concrete class (all abstract members `type`, `hasNext`, and `next` have implementations):
+
 ```tut
```

Comment:
I am not sure. I am not apposed to it. Maybe a link to another part of the tour that explains the semantics of  `private[this] var...` would make it acceptable.